### PR TITLE
Tasks

### PR DESCRIPTION
**Pull request by vladmet1**
function var. 14

### CODE REVIEW

File: функция.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿namespace CourseApp
+{
+    using System;
+
+    public class Функция
```
Comment (alexthvest) [inline функция.cs:5]: Please, do not use Russian in naming

File: функция.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    using System;
+
+    public class Функция
+    {
+        public void Calculate(double a, double b, double x)
+        {
+            double top = Math.Sqrt(Math.Abs(a - (b * x)));
+            double bottom = Math.Sqrt(Math.Pow(Math.Log(10), 3));
+            double y = Math.Round(top / bottom, 3);
```
Comment (alexthvest) [inline функция.cs:11]: Do round only on output

File: функция.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            Console.WriteLine(y);
```
Comment (alexthvest) [inline функция.cs:12]: Method should do only one thing - calculation

**Review by alexthvest — CHANGES_REQUESTED** [review]

File: функция.cs (hunk 2 above)
Comment (vladmet1) [reply функция.cs:11]: double y = top / bottom;
 Console.WriteLine(Math.Round(y, 3));

Ok?

File: функция.cs (hunk 1 above)
Comment (vladmet1) [reply функция.cs:5]: I'll fix it

File: функция.cs (hunk 2 above)
Comment (alexthvest) [reply функция.cs:11]: Yes, but it will be good if you move output outside this method

File: функция.cs (hunk 1 above)
Comment (jskonst) [reply функция.cs:5]: 👁️ 🩸

File: DemoTest.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        [Fact]
+        public void Test()
+        {
+            Assert.True(true);
+        }
+
+        [Fact]
+        public void TestIntSum()
+        {
```
Comment (jskonst) [inline DemoTest.cs:15]: Please - use inlineData

File: Function.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    using System;
+
+    public class Function
+    {
+        public void Calculate(double a, double b, double x)
+        {
+            double top = Math.Sqrt(Math.Abs(a - (b * x)));
+            double bottom = Math.Sqrt(Math.Pow(Math.Log(10), 3));
+            double y = top / bottom;
+            Console.WriteLine(Math.Round(y, 3));
```
Comment (jskonst) [inline Function.cs:12]: this one should return valuse

File: Function.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
+namespace CourseApp
+{
+    using System;
+
+    public class Function
+    {
+        public void Calculate(double a, double b, double x)
```
Comment (jskonst) [inline Function.cs:7]: Please - add tests for this one

File: ProgramDish.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
+﻿namespace CourseApp1
+{
+    using System;
+
+    public class ProgramDish
```
Comment (jskonst) [inline ProgramDish.cs:5]: Additionally - we need tests for this file

**Review by jskonst — CHANGES_REQUESTED** [review]

File: DishClass.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        public object ProgramDish(string name, int weight, int price)
+        {
+            Console.WriteLine("dish constructor called");
+            Name = name;
+            Weight = weight;
+            Price = price;
+            return (name, weight, price);
+        }
```
Comment (alexthvest) [inline DishClass.cs:36]: Nice try :) But it is not constructor
Constructor name = Class name and has no return value

File: DishClass.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
+namespace CourseApp
+{
+    using System;
+
+    public class DishClass
```
Comment (alexthvest) [inline DishClass.cs:5]: Please do not use `Class` postfix in class names, we already know what is it

File: DemoTest.cs (hunk 4 above)
Comment (jskonst) [reply DemoTest.cs:15]: @vladmet1 Please - fix review points

File: CalculatorTest.cs [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+            var res = calc.GetSum(a, b);
+
+            Assert.Equal(exp, res, 3);
+        }
+
+        [Theory]
+        [InlineData(10, 5, 15)]
+        [InlineData(40.123675f, 29.812398f, 72.936073f)]
+        [InlineData(67.123, 3.002, 70.125)]
+        public void NestCalc3(double a, double b, double exp)
```
Comment (jskonst) [inline CalculatorTest.cs:37]: Nest?

Comment (jskonst) [reply CalculatorTest.cs:37]: ans what about test case where a or b equals to 0 ?

File: DemoTest.cs [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+            Assert.Equal(expected, res);
+        }
+
+        [Theory]
+        [InlineData(5, 2, 2.5)]
+        [InlineData(4.4, 2.2, 2)]
+        [InlineData(0.25, 0.2, 1.25)]
+        [InlineData(10, 5, 2)]
+        [InlineData(1.28, 0.8, 1.6)]
+        public void TestDoubleQuotient(int firstNumber, int secondNumber, int expected)
```
Comment (jskonst) [inline DemoTest.cs:39]: Please add test for the case where a or (and) B are equal to 0

File: Dish.cs [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+            set
+            {
+                if (value > 0 & value < 10000)
+                {
+                    price = value;
+                }
+            }
+        }
+
+        public object ProgramDish(string name, int weight, int price)
```
Comment (jskonst) [inline Dish.cs:29]: why object?

File: Dish.cs [hunk 13]
```diff
@@ -0,0 +1,38 @@
+namespace CourseApp
+{
+    using System;
+
+    public class Dish
+    {
+        private int price;
```
Comment (jskonst) [inline Dish.cs:7]: For Dish we need at least 2 Constructors

File: Dish.cs [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+                if (value > 0 & value < 10000)
+                {
+                    price = value;
+                }
+            }
+        }
+
+        public object ProgramDish(string name, int weight, int price)
+        {
+            Console.WriteLine("dish constructor called");
```
Comment (jskonst) [inline Dish.cs:31]: i think -that it's not the buiseness of this method to write something to the console

File: CalculatorTest.cs [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
+            var res = calc.GetSum(a, b);
+
+            Assert.Equal(exp, res, 3);
+        }
+
+        [Theory]
+        [InlineData(10, 5, 15)]
+        [InlineData(40.123675f, 29.812398f, 72.936073f)]
+        [InlineData(67.123, 3.002, 70.125)]
```
Comment (jskonst) [inline CalculatorTest.cs:36]: Please check case with 0 for numerator and enumerator

**Pull request closed without merging**